eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -62,49 +62,46 @@ function create-metadata-table($readmeFolder, $readmeName, $moniker, $msService,
+  $readmeHeader = "# Azure $serviceName SDK for $languageDisplayName - $moniker`r`n"
   Add-Content -Path $readmePath -Value $readmeHeader
-  Add-Content -Path $readmePath -Value $content
-}
-
-function compare-and-merge-metadata ($original, $updated) {
-  $originalTable = ConvertFrom-StringData -StringData $original -Delimiter ":"
-  $updatedTable = ConvertFrom-StringData -StringData $updated -Delimiter ":"
-  foreach ($key in $originalTable.Keys) {
-    if (!($updatedTable.ContainsKey($key))) {
-      Write-Warning "New metadata missed the entry: $key. Adding back."
-      $updated += "$key`: $($originalTable[$key])`r`n"
-    }
-  }
-  return $updated
+  Add-Content -Path $readmePath -Value $content -NoNewline
 }
 
 # Update the metadata table.
 function update-metadata-table($readmeFolder, $readmeName, $serviceName, $msService)
 {
   $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
   $readmeContent = Get-Content -Path $readmePath -Raw
-  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
-  $restContent = $Matches["content"]
-  $orignalMetadata = $Matches["metadata"]
+  $match = $readmeContent -match "^---\n*(?<metadata>(.*\n?)*?)---\n*(?<content>(.*\n?)*)"
+  if (!$match) {
+    # $Language, $LanguageDisplayName are the variables globally defined in Language-Settings.ps1
+    $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
+      -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+      -msService $msService
+    Set-Content -Path $readmePath -Value "$metadataString$readmeContent" -NoNewline
+    return
+  }
+  $restContent = $Matches["content"].trim()
+  $metadata = $Matches["metadata"].trim()
   # $Language, $LanguageDisplayName are the variables globally defined in Language-Settings.ps1
-  $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
+  $metadataString = GenerateDocsMsMetadata -originalMetadata $metadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
     -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
     -msService $msService
-  $null = $metadataString -match "---`n*(?<metadata>(.*`n)*)---"
-  $mergedMetadata = compare-and-merge-metadata -original $orignalMetadata -updated $Matches["metadata"]
-  Set-Content -Path $readmePath -Value "---`n$mergedMetadata---`n$restContent" -NoNewline
+  Set-Content -Path $readmePath -Value "$metadataString$restContent" -NoNewline
 }
 
 function generate-markdown-table($readmeFolder, $readmeName, $packageInfo, $moniker) {
   $tableHeader = "| Reference | Package | Source |`r`n|---|---|---|`r`n" 
   $tableContent = ""
   # Here is the table, the versioned value will
   foreach ($pkg in $packageInfo) {
-    $repositoryLink = $RepositoryUri
+    $repositoryLink = "$RepositoryUri/$($pkg.Package)"
```

Comment:
This is not a good default as we do not have this value officially at the time read this line.
Prefer to leave this empty for future change.